src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1235,21 +1234,6 @@ impl EventProcessor {
         );
 
         self.update_mods_from_query(window_id, &mut callback);
-
-        // The deviceid for this event is for a keyboard instead of a pointer,
-        // so we have to do a little extra work.
-        let pointer_id = self
-            .devices
-            .borrow()
-            .get(&DeviceId(xev.deviceid as xinput::DeviceId))
-            .map(|device| device.attachment)
-            .unwrap_or(2);
-
-        let event = Event::WindowEvent {
-            window_id,
-            event: WindowEvent::CursorMoved { device_id: mkdid(pointer_id as _), position },
-        };
```

Comment:
I tracked this down to #348.

While I do think this is a valid use-case, the current implementation is incorrect, because its also emits an event if the cursor is not inside the window.

For this to be solved we need to do it across all platforms and do it correctly.
My proposition would be to emit a `CursorEntered` when focus is gained _if_ the cursor is inside the window.

On the other hand this might also be addressed by https://github.com/rust-windowing/winit/pull/2648.